sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -95,102 +115,148 @@ function createPropertiesFromSpan(span: ReadableSpan): [Properties, Measurements
+}
+
+function getTarget(span: ReadableSpan): string {
+  const peerService = span.attributes[SemanticAttributes.PEER_SERVICE];
+  const httpHost = span.attributes[SemanticAttributes.HTTP_HOST];
+  const httpUrl = span.attributes[SemanticAttributes.HTTP_URL];
+  const netPeerName = span.attributes[SemanticAttributes.NET_PEER_NAME];
+  const netPeerIp = span.attributes[SemanticAttributes.NET_PEER_IP];
+  if (peerService) {
+    return String(peerService);
+  } else if (httpHost) {
+    return String(httpHost);
+  } else if (httpUrl) {
+    return String(httpUrl);
+  } else if (netPeerName) {
+    return String(netPeerName);
+  } else if (netPeerIp) {
+    return String(netPeerIp);
+  }
+  return "";
+}
+
 function createDependencyData(span: ReadableSpan): RemoteDependencyData {
-  const data: RemoteDependencyData = {
+  const remoteDependencyData: RemoteDependencyData = {
     name: span.name,
     id: `|${span.spanContext().traceId}.${span.spanContext().spanId}.`,
-    success: span.status.code === SpanStatusCode.OK,
-    resultCode: String(span.status.code),
-    target: span.attributes[HTTP_URL] as string | undefined,
+    success: span.status.code != SpanStatusCode.ERROR,
+    resultCode: "0",
     type: "Dependency",
     duration: msToTimeSpan(hrTimeToMilliseconds(span.duration)),
-    version: 1
+    version: 2
   };
-
-  if (span.attributes[HTTP_STATUS_CODE]) {
-    data.type = "HTTP";
-    data.resultCode = String(span.attributes[HTTP_STATUS_CODE]);
-  }
-
-  if (span.attributes[GRPC_STATUS_CODE] !== undefined) {
-    data.type = "GRPC";
-    data.resultCode = String(span.attributes[GRPC_STATUS_CODE]);
-  }
-
-  if (span.attributes[GRPC_METHOD]) {
-    data.target = String(span.attributes[GRPC_METHOD]);
-    data.data = String(span.attributes[GRPC_METHOD]);
+  const httpMethod = span.attributes[SemanticAttributes.HTTP_METHOD];
+  const dbSystem = span.attributes[SemanticAttributes.DB_SYSTEM];
+  const rpcMethod = span.attributes[SemanticAttributes.RPC_METHOD];
+  // HTTP Dependency
+  if (httpMethod) {
+    remoteDependencyData.type = DependencyTypes.Http;
+    remoteDependencyData.data = getUrl(span);
+    const httpStatusCode = span.attributes[SemanticAttributes.HTTP_STATUS_CODE];
+    if (httpStatusCode) {
+      remoteDependencyData.resultCode = String(httpStatusCode);
+    }
+    let target = getTarget(span);
+    if (target) {
+      let url = new URL(target);
+      // Ignore port if it is a default port
+      if (url.port === "80" || url.port === "443") {
+        url.port = "";
+      }
+      remoteDependencyData.target = `${url}`;
+    }
   }
-
-  if (span.attributes[HTTP_URL]) {
-    const url = new URL(span.attributes[HTTP_URL] as string);
-    data.target = url.hostname;
-    data.data = url.href;
-
-    if (span.attributes[HTTP_METHOD]) {
-      data.name = `${span.attributes[HTTP_METHOD] as string} ${url.pathname}`;
+  // DB Dependency
+  else if (dbSystem) {
+    remoteDependencyData.type = String(dbSystem);
```

Comment:
Shouldn't this be `SQL` if relational? https://github.com/microsoft/common-schema/blob/main/Mappings/AzureMonitor-AI.md#remotedependencytype